src/hammer-vlsi/hammer_vlsi/hammer_build_systems.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  hammer_build_systems.py
+#  Class containing all the methods to create VLSI flow build system infrastructure
+#
+#  See LICENSE for licence details.
+
+from .driver import HammerDriver
+
+import os
+import sys
+import textwrap
+from typing import List, Dict, Tuple, Callable
+
+def build_noop(driver: HammerDriver) -> dict:
+    """
+    Do nothing, just return the dependency graph.
+
+    :param driver: The HammerDriver
+    :return: The dependency graph
+    """
+    dependency_graph = driver.get_hierarchical_dependency_graph()
+    return dependency_graph
+
+
+def build_makefile(driver: HammerDriver) -> dict:
+    """
+    Build a Makefile include in the obj_dir called hammer.d. This is intended to be dynamically
+    created and included into a top-level Makefile.
+
+    :param driver: The HammerDriver
+    :return: The dependency graph
+    """
+    dependency_graph = driver.get_hierarchical_dependency_graph()
+    makefile = os.path.join(driver.obj_dir, "hammer.d")
+    default_dependencies = driver.options.project_configs + driver.options.environment_configs
+    default_dependencies.extend(list(driver.database.get_setting("synthesis.inputs.input_files", [])))
```

Comment:
Is there some way to get add to this dynamically? I'm thinking of transcluded files for instance.

Reply:
Yikes, probably, but that sounds really hard.

Reply:
Filed #455